tests/test_pagination.py

```diff
@@ -0,0 +1,52 @@
+from tap_tester import runner, connections
+
+from base import TestLinkedinAdsBase
+
+class LinkedinAdsPaginationTest(TestLinkedinAdsBase):
+
+    @staticmethod
+    def name():
+        return "tap_tester_linkedin_ads_pagination_test"
+
+    def test_run(self):
+        # page size for "campaign_groups" set in the tap
+        page_size = 100
+        conn_id = connections.ensure_connection(self)
+
+        # Checking pagination for "campaign_groups" stream
+        expected_streams = ["campaign_groups"]
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # table and field selection
+        test_catalogs = [catalog for catalog in found_catalogs
+                                      if catalog.get('stream_name') in expected_streams]
+
+        self.perform_and_verify_table_and_field_selection(conn_id, test_catalogs)
+
+        record_count_by_stream = self.run_and_verify_sync(conn_id)
+
+        synced_records = runner.get_records_from_target_output()
+
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+                # expected values
+                expected_primary_keys = self.expected_primary_keys()
+
+                # collect information for assertions from syncs 1 & 2 base on expected values
```

Comment:
Minor issue with the comment - there is only 1 sync performed in this test.

Reply:
Done